kotlintest-runner/kotlintest-runner-jvm/src/main/kotlin/io/kotlintest/runner/jvm/TestCaseExecutor.kt

```diff
@@ -160,16 +162,27 @@ class TestCaseExecutor(private val listener: TestEngineListener,
    */
   private fun before(testCase: TestCase) {
     listener.enterTestCase(testCase)
-    val userListeners = listOf(testCase.spec) + testCase.spec.listeners() + Project.listeners()
+
+    val userListeners = testCase.spec.listeners() + Project.listeners()
     userListeners.forEach { it.beforeTest(testCase.description) }
+
+    if (testCase.config.enabled) {
+      // Only execute before test from the spec if the test is enabled
```

Comment:
Let's remove this functionality from here. It has nothing to do with coroutines.
We'll address it in the "rework listeners" branch.

Reply:
It has to do with coroutines as this is part of merging from master branch. This is already present in master branch